server/treetest_server.js

```diff
@@ -130,9 +139,17 @@ module.exports = {
         });
     },
     update: function (req, res, next) {
-        var tasks = req.body.tasks.split(/\r?\n/).map(function(item) {
-             return item.trim();
-        }).filter(function(n){ return n != '' });
+        var tasks = [];
+        for (var i = 0; i < req.body.question.length; i++) {
+            if (req.body.question[i] == '') {
```

Comment:
Is there a need for triming?
```suggestion
            if (req.body.question[i].trim() == '') {
```

What does that mean when a question is empty in practice?

Reply:
Yes, good to add trim to ensure we are taking into account whitespaces as well. I will also add some comments around update function.

In practice, we should never have a empty question. This check just ensures that in case a user clicks on `New Task` button by mistake, empty question is not saved to the database. Actual tree test uses questions stored in the database to prompt the users, and we do not want to confuse them.

Reply:
This woul results in tasks having missing indices as well. Does that have any unwanted effect in the application, or is it required for it to work?
Otherwise we could use `push()` to avoid skipping indices.

Reply:
No it shouldn't since we are using `tasks[tasks.length] = ...` which should always use the next index. But I can change to use `push()` if that makes it more clearer.